tcms/testruns/tests/test_views.py

```diff
@@ -444,15 +445,75 @@ def setUpTestData(cls):
         cls.case_run_1.case.text = "Scenario Version 2"
         cls.case_run_1.case.save()
 
-    def test_update_selected_case_runs(self):
+        cls.update_case_run_text_html = \
+            '<a href="#" title="Update the IDLE case runs to newest case text" ' \
+            'href="javascript:void(0)" data-param="{0}" ' \
+            'class="updateBlue9 js-update-case">{1}</a>' \
+            .format(
+                reverse('testruns-update_case_run_text', args=[cls.test_run.pk]),
+                _('Update')
+            )
+
+    def test_get_update_caserun_text_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.testruns_url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.update_case_run_text_html, html=True)
+
+    def test_update_selected_case_runs_with_permissions(self):
+        user_should_have_perm(self.tester, 'testruns.change_testcaserun')
+
         self.assertNotEqual(self.case_run_1.case.history.latest().history_id,
                             self.case_run_1.case_text_version)
         response = self.client.post(self.update_url,
                                     {'case_run': [self.case_run_1.pk]},
                                     follow=True)
 
         self.assertContains(response, _('%d CaseRun(s) updated:') % 1)
-        self.assertEqual(self.case_run_1.case.text, "Scenario Version 2")
+
+        self.case_run_1.refresh_from_db()
+
+        self.assertEqual(
+            self.case_run_1.case.get_text_with_version(
+                self.case_run_1.case_text_version
+            ),
+            "Scenario Version 2"
+        )
+        self.assertEqual(
+            self.case_run_1.case.history.latest().history_id,
+            self.case_run_1.case_text_version
+        )
+
+    def test_get_update_caserun_text_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+        response = self.client.get(self.testruns_url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertNotContains(response, self.update_case_run_text_html, html=True)
+
+    def test_update_selected_case_runs_without_permissions(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testcaserun')
+
+        self.assertEqual(self.case_run_1.case.history.latest().history_id,
```

Comment:
this should be `assertNotEqual` b/c when we first try to update the text whatever we have inside the test run is already older. So the test makes sure we are starting with a TestCase object which has some updates already.